src/goto-instrument/goto_instrument_parse_options.cpp

```diff
@@ -694,7 +694,14 @@ int goto_instrument_parse_optionst::doit()
       goto_inline(goto_model, get_message_handler());
 
       status() << "Accelerating" << eom;
-      accelerate_functions(goto_model, cmdline.isset("z3"));
+      optionst options;
+      if(cmdline.isset("random-seed"))
+        options.set_option("random-seed", cmdline.get_value("random-seed"));
+      if(cmdline.isset("random-var-freq"))
+        options.set_option(
+          "random-var-freq", cmdline.get_value("random-var-freq"));
+      accelerate_functions(
+        goto_model, cmdline.isset("z3"), parse_satcheck_info(options));
```

Comment:
This is a weird way of doing that. I suggest to pass optionst only and sort out the mess inside accelerate_functions.